src/python/pants/option/custom_types.py

```diff
@@ -100,6 +100,18 @@ def shell_str(s: str) -> str:
     """
     return s
 
+def workspace_path(s: str) -> str:
+    """Same type as 'str', but indicates string represents a directory path that is relative to
+    either the build root, or a BUILD file if prefix with `./`.
+
+    :API: public
+    """
+    if s.startswith("/"):
+        raise ParseError(
+            f"Invalid value: `{s}`. Expected a relative path, optionally in the form "
+            "`./relative/path` to make it relative to the BUILD files rather than the build root."
+        )
+    return s
```

Comment:
Copied from https://github.com/pantsbuild/pants/blob/3e9d83061cbc1b043c9e409c9b2485df09e8de1a/src/python/pants/backend/docker/subsystems/docker_options.py

I should remove that code :facepalm: